File: dart/receive.dart
import "dart:io";
import "package:dart_amqp/dart_amqp.dart";

void main (List<String> arguments) {
    ConnectionSettings settings = new ConnectionSettings(
        host: "localhost"
    );

    Client client = new Client(settings: settings);

    ProcessSignal.sigint.watch().listen((_) {
        client.close().then((_) {
            print("close client");
            exit(0);
        });
    });

    String msg = arguments.isEmpty ? "Hello World!": arguments[0];

    String queueTag = "hello";

    client
        .channel()
        .then((Channel channel) => channel.queue(queueTag, durable: false))
        .then((Queue queue) {
            print(" [*] Waiting for messages in ${queueTag}. To Exit press CTRL+C");
            return queue.consume(consumerTag: queueTag, noAck: true);
        })
        .then((Consumer consumer) {
            consumer.listen((AmqpMessage event) {
                print(" [x] Received ${event.payloadAsString}");
            });
        });
}

File: julia/receive.jl
using AMQPClient
const VIRTUALHOST = "/"
const HOST = "127.0.0.1"


function receive()
    # 1. Create a connection to the localhost or 127.0.0.1 of virtualhost '/'
    connection(; virtualhost=VIRTUALHOST, host=HOST) do conn
        println("Connection established")
        # 2. Create a channel to send messages
        channel(conn, AMQPClient.UNUSED_CHANNEL, true) do chan
            # 3. Declare a queue
            println("Channel created")
            queue = "hello"
            success, queue_name, message_count, consumer_count = queue_declare(chan, queue)

            # 4. Setup to receive
            on_receive = (msg) -> begin
                # 4.1 Receive message is Vector{UInt8}
                data = String(msg.data)
                println("Received the message: $data")
                # 4.2 Acknowledge the message
                basic_ack(chan, msg.delivery_tag)
            end

            success, consumer_tag = basic_consume(chan, queue, on_receive)
            @assert success == true

            # 5. Run for-ever
            # listen to new messages
            while true
                sleep(1)
            end
        end
    end
end

# Don't exit on Ctrl-C
Base.exit_on_sigint(false)
try
    receive()
catch ex
    if ex isa InterruptException
        println("Interrupted")
    else
        println("Exception: $ex")
    end
end
